### ISSUE TITLE
Recommended safe way to create secrets in Secret Manger

### ISSUE THREAD

**Issue by kaitoii11** [GoogleCloudPlatform/k8s-config-connector#817]
### Describe your question

I noticed that I can create [secret manager](https://cloud.google.com/config-connector/docs/reference/resource-docs/secretmanager/secretmanagersecret) in Config Connector but I was curious as to how one should create the secrets.

Is there a recommended way to create secrets using Config Connector? Or do I have to manually add through GCP console or gcloud CLI?

**Comment by diviner524** [comment]
@kaitoii11 Are you looking for SecretManagerSecretVersion or something different?

https://cloud.google.com/config-connector/docs/reference/resource-docs/secretmanager/secretmanagersecretversion#typical_use_case

**Comment by tcaky** [comment]
Hi, I wanted to jump in on this.  The secretmanagersecretversion typical use case relies on the secret pre-existing.  Would it not make sense to have an option in the spec for something like `autocreatesecret: true`?  I am running into the issue from an IaC perspective when creating a Postgres instance and DB.  Ideally there would be a completely hands-off way to do this.

When I used to be an admin for SharePoint there was a concept of SharePoint-Managed Accounts, where you could have SharePoint take over password rotation on a time frame you could describe such as monthly, 90 days, yearly, etc.  I think there could be value in such a thing for secretmanager.

**Comment by diviner524** [comment]
@tcaky I agree this is a valid feature request. The best option would be to have the secret version feature in GCP (and the corresponding underlying API) support auto creation of secrets. This will ensure that the secret value is safely stored in the cloud instead of being stored as a Kubernetes secret.

Once this feature is available, it can be added to Config Connector as a new field in the spec. Unfortunately it seems this is not supported in the secret version doc. [1].

[1] https://cloud.google.com/secret-manager/docs/add-secret-version

**Comment by tcaky** [comment]
> @tcaky I agree this is a valid feature request. The best option would be to have the secret version feature in GCP (and the corresponding underlying API) support auto creation of secrets. This will ensure that the secret value is safely stored in the cloud instead of being stored as a Kubernetes secret.
> 
> Once this feature is available, it can be added to Config Connector as a new field in the spec. Unfortunately it seems this is not supported in the secret version doc. [1].
> 
> [1] https://cloud.google.com/secret-manager/docs/add-secret-version

@diviner524  Would you be able to direct me to where I could submit this feature request?

**Comment by ldanielmadariaga** [comment]
This item has been open for >12 months. As part of scheduled repository maintenance, this Issue/PR will be closed. If this is still a critical ask, please re-open and work with your Google Cloud Account team to prioritize it with GCP KCC Product Management.
